.github/workflows/install-slices.yaml

```diff
@@ -0,0 +1,121 @@
+          ]]; then
+            echo "matrix={\"ref\":${{ env.RELEASES }},\"arch\":${{ env.ARCHES }}}" >> $GITHUB_OUTPUT
+            echo "install_all=true" >> $GITHUB_OUTPUT
+          else
+            echo "matrix={\"arch\":${{ env.ARCHES }}}" >> $GITHUB_OUTPUT
+          fi
+
+  # The "install" job tests the slices by installing them.
+  # It installs **all** slices if:
+  #   - inputs.install-all is true
+  #   - chisel.yaml is changed
+  #   - any slice definition files are deleted
+  #   - github workflows (and related files) are changed
+  # Otherwise, it installs only the slices from **added** and/or
+  # **modified** slice definition files.
+  # Please note that "change" is defined to encompass additions,
+  # modifications and deletions here.
+  install:
+    runs-on: ubuntu-latest
+    name: "Install"
+    needs: prepare-install
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.prepare-install.outputs.matrix) }}
+    env:
+      install-all: ${{ needs.prepare-install.outputs.install-all }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ matrix.ref }}
+
+      - name: Check changed paths
+        id: changed-paths
+        if: env.install-all != 'true'
+        uses: dorny/paths-filter@v3
+        with:
+          # ref: https://github.com/marketplace/actions/paths-changes-filter
+          filters: |
+            install-all:
+              - 'chisel.yaml'
+              - deleted: 'slices/**/*.yaml'
+              - '.github/**'
+            slices:
+              - added|modified: 'slices/**/*.yaml'
+          # Space delimited list usable as command-line argument list in
+          # Linux shell. If needed, it uses single or double quotes to
+          # wrap filename with unsafe characters.
+          list-files: shell
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+
+      - name: Install dependencies
+        run: |
+          go install github.com/canonical/chisel/cmd/chisel@latest
+          go install github.com/mikefarah/yq/v4@latest
+          wget -q https://raw.githubusercontent.com/canonical/chisel-releases/main/.github/scripts/install-slices
+          chmod +x install-slices
+
+      - name: Install all slices
+        if: |
+          env.install-all == 'true' ||
+          steps.changed-paths.outputs.install-all == 'true'
+        run: |
+          set -e
+          # we need to enable globstar to use the ** patterns below
+          shopt -s globstar
+          ./install-slices --arch "${{ matrix.arch }}" --release ./ \
+            slices/**/*.yaml
+
+      - name: Install slices from changed files
+        if: |
+          steps.changed-paths.outputs.slices == 'true' &&
+          !(
+            env.install-all == 'true' ||
+            steps.changed-paths.outputs.install-all == 'true'
+          )
+        run: |
+          set -e
+          ./install-slices --arch "${{ matrix.arch }}" --release ./ \
```

Comment:
since the `run` is pretty much the same in these two last steps, you might want to try to merge them, and handle the conditions in shell, to avoid duplication

Reply:
Thanks, merged in https://github.com/canonical/chisel-releases/pull/119/commits/71a897eea58ac17332275feb6b252f1fb5d0931d.